R/clean-wqdata.R

```diff
@@ -24,48 +24,48 @@ abs_dev <- function(x) {
   abs(x - mean(x))
 }
 
-clean_wqdata_replicates <- function(x, max_cv, messages) {
+clean_wqdata_replicates <- function (x, max_cv, messages, FUN) {
   n <- nrow(x)
   cv <- cv(x$Value)
   if (cv(x$Value) > max_cv && nrow(x) > 2) {
-    x <- dplyr::arrange_(x, ~ -Value)
+    x <- dplyr::arrange_(x, ~-Value)
     while (cv(x$Value) > max_cv && nrow(x) > 2) {
       x <- x[-which.max(abs_dev(x$Value)), ]
     }
   }
-  x$Value <- mean(x$Value)
 
-  if (!is.null(x$DetectionLimit)) {
-    x$DetectionLimit <- mean(x$DetectionLimit)
+  if(FUN == "mean"){
+    x$Value <- mean(x$Value)
   }
-
+  if(FUN == "max"){
+    x$Value <- max(x$Value)
+  }
+  if(FUN == "median"){
+    x$Value <- stats::median(x$Value)
+  }
+  if (!is.null(x$DetectionLimit))
+    x$DetectionLimit <- mean(x$DetectionLimit)
   if (messages && n > nrow(x)) {
-    message(
-      "Filtered ", n - nrow(x), " of ", n,
-      " replicate values with a CV of ", signif(cv, 3), " for ", x$Variable[1],
-      " on ", x$Date[1], "."
-    )
+    message("Filtered ", n - nrow(x), " of ", n, " replicate values with a CV of ",
+            signif(cv, 3), " for ", x$Variable[1], " on ", x$Date[1],
+            ".")
   }
   x[1, , drop = FALSE]
 }
 
-clean_wqdata_variable <- function(x, max_cv, messages) {
-  if (anyDuplicated(x$Date)) {
+clean_wqdata_variable <- function (x, max_cv, messages, FUN) {
+  if (anyDuplicated(x$Date))
```

Comment:
Nit: here and in the next block, I prefer curly braces around the expressions following `if` when there are multiple lines

Reply:
I agree that passing a function is more elegant. However, I did this (along with argument name) to stay consistent with other functions in the package...e.g. `summarise_for_trends()` and `test_trends()`. I'm open to either, depending on whether you think consistency is particularly important

Reply:
yes I will change so accepts a function instead of the name of it